src/plopp/widgets/linecut_tool.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from ..core import node, Node, View
+from ..core.utils import coord_as_midpoints
+from ..graphics.fig2d import Figure2d
+from .tools import DrawLinesTool
+
+from functools import partial
+import numpy as np
+import scipp as sc
+from scipy.interpolate import interp2d
+from typing import Any, Dict, Optional
+
+
+def interpolator(xy, interp_func, resolution, data_unit, distance_unit):
+    x1 = xy['x'][0]
+    y1 = xy['y'][0]
+    x2 = xy['x'][1]
+    y2 = xy['y'][1]
+
+    xvalues = np.linspace(x1, x2, resolution)
+    yvalues = np.linspace(y1, y2, resolution)
+
+    profile = np.array([interp_func(x, y)[0] for x, y in zip(xvalues, yvalues)])
+    out = sc.DataArray(data=sc.array(dims=['distance'], values=profile, unit=data_unit),
+                       coords={
+                           'distance':
+                           sc.array(dims=['distance'],
+                                    values=np.linalg.norm(
+                                        [xvalues - xvalues[0], yvalues - yvalues[0]],
+                                        axis=0),
+                                    unit=distance_unit)
```

Comment:
I don't think this a useful coordinate in practice? I can imagine someone might use this on an image, or a I(Qx, Qy, Qz) intensity, but in either case the user would want to know the coordinates.